lib/fizzy/capi.cpp

```diff
@@ -404,65 +405,66 @@ const FizzyModule* fizzy_clone_module(const FizzyModule* module)
     }
 }
 
-uint32_t fizzy_get_type_count(const FizzyModule* module)
+uint32_t fizzy_get_type_count(const FizzyModule* module) noexcept
 {
     return static_cast<uint32_t>(unwrap(module)->typesec.size());
 }
 
-FizzyFunctionType fizzy_get_type(const FizzyModule* module, uint32_t type_idx)
+FizzyFunctionType fizzy_get_type(const FizzyModule* module, uint32_t type_idx) noexcept
 {
     return wrap(unwrap(module)->typesec[type_idx]);
 }
 
-uint32_t fizzy_get_import_count(const FizzyModule* module)
+uint32_t fizzy_get_import_count(const FizzyModule* module) noexcept
 {
     return static_cast<uint32_t>(unwrap(module)->importsec.size());
 }
 
 FizzyImportDescription fizzy_get_import_description(
-    const FizzyModule* c_module, uint32_t import_idx)
+    const FizzyModule* c_module, uint32_t import_idx) noexcept
 {
     const auto* module = unwrap(c_module);
     return wrap(module->importsec[import_idx], *module);
 }
 
-FizzyFunctionType fizzy_get_function_type(const FizzyModule* module, uint32_t func_idx)
+FizzyFunctionType fizzy_get_function_type(const FizzyModule* module, uint32_t func_idx) noexcept
 {
     return wrap(unwrap(module)->get_function_type(func_idx));
 }
 
-bool fizzy_module_has_table(const FizzyModule* module)
+bool fizzy_module_has_table(const FizzyModule* module) noexcept
 {
     return unwrap(module)->has_table();
 }
 
-bool fizzy_module_has_memory(const FizzyModule* module)
+bool fizzy_module_has_memory(const FizzyModule* module) noexcept
 {
     return unwrap(module)->has_memory();
 }
 
-uint32_t fizzy_get_global_count(const FizzyModule* module)
+uint32_t fizzy_get_global_count(const FizzyModule* module) noexcept
 {
     return static_cast<uint32_t>(unwrap(module)->get_global_count());
 }
 
-FizzyGlobalType fizzy_get_global_type(const FizzyModule* module, uint32_t global_idx)
+FizzyGlobalType fizzy_get_global_type(const FizzyModule* module, uint32_t global_idx) noexcept
 {
     return wrap(unwrap(module)->get_global_type(global_idx));
 }
 
-uint32_t fizzy_get_export_count(const FizzyModule* module)
+uint32_t fizzy_get_export_count(const FizzyModule* module) noexcept
 {
     return static_cast<uint32_t>(unwrap(module)->exportsec.size());
 }
 
-FizzyExportDescription fizzy_get_export_description(const FizzyModule* module, uint32_t export_idx)
+FizzyExportDescription fizzy_get_export_description(
+    const FizzyModule* module, uint32_t export_idx) noexcept
 {
     return wrap(unwrap(module)->exportsec[export_idx]);
 }
 
 bool fizzy_find_exported_function_index(
-    const FizzyModule* module, const char* name, uint32_t* out_func_idx)
+    const FizzyModule* module, const char* name, uint32_t* out_func_idx) noexcept
 {
     const auto optional_func_idx = fizzy::find_exported_function(*unwrap(module), name);
```

Comment:
`fizzy::find_exported_function` is not marked `noexcept`, I'm not sure exactly why, perhaps because `std::make_optional` in it is not `noexcept` (it can throw in case underlying object's contructor throws, which is not our case, we have `optional<uint32_t>`)

I would expect here that clang-tidy would complain that `fizzy_find_exported_function_index` can't be made `noexcept` either... not sure why it doesn't.

Reply:
I think we should try making `find_exported_*` functions on C++ side `noexcept`.

Reply:
Fixed by #776.